### PR TITLE
Issues supporting aws privatelink

### PR DESCRIPTION
**Pull request by egor-krv**
This PR solves this [issue](https://github.com/ClickHouse/terraform-provider-clickhouse/issues/30). To solve the issue we need to have a ClickHouse service provisioned first before attaching the endpoints because the endpoints creation depends on the ClickHouse service itself. I moved endpoint attachment to a new resource so it happens after the endpoints are created.

### CODE REVIEW

File: examples/PrivateLink/main.tf [hunk 1]
```diff
@@ -1,7 +1,7 @@
 terraform {
   required_providers {
     clickhouse = {
-      version = "0.0.5"
+      version = "0.0.10"
```
Comment (whites11) [inline examples/PrivateLink/main.tf:4]: technically this example won't work until this PR is merged, and thus released with a future version of the provider, right?

File: examples/PrivateLink/aws.tf [hunk 2]
```diff
@@ -1,19 +1,9 @@
...
-  type = string
-}
-
 variable "aws_region" {
   type = string
 }
 
 provider "aws" {
   region     = var.aws_region
-  access_key = var.aws_key
```
Comment (whites11) [inline examples/PrivateLink/aws.tf:15]: How is this change related with this PR?

File: examples/PrivateLink/main.tf (hunk 1 above)
Comment (egor-krv) [reply examples/PrivateLink/main.tf:4]: This is the current version so we are all good here.

File: examples/PrivateLink/aws.tf (hunk 2 above)
Comment (egor-krv) [reply examples/PrivateLink/aws.tf:15]: Just modifying the example so it works from environment vars so it is easier to run

**Review by whites11 — APPROVED** [review]
